### ISSUE TITLE
[TOURS] Testen van de pagina

### ISSUE THREAD

**Issue by zainablfz** [Patrickkhr/release-candidate#45]
- [x] Het testen van de pagina op responsiveness, accessibility, progressive enhancement, performance & testresultaten verwerken

-  Responsive Design test
-  User test
-  W3C HTML Validator
-  Accessibility test
  -  Lighthouse test
  -  Color Contrast Test
  -  Screenreader
  -  Keyboard / Interactive Elements Test
-  Performance test
  -  Lighthouse test
-  Progressive Enhancement:
  -  Disabling JavaScript 
  -  Device test
  -  Browser test
  -  Safari, Chrome & Firefox

**Comment by zainablfz** [comment]
# Testen 

Na het bouwen van mijn pagina ben ik begonnen met de testfase. Dit heb ik gedaan door middel van verschillende testen en heb ik deze resultaten verwerkt in de code. Hierbij heb ik rekening gehouden met responsiveness, accessibility, performance & progressive enhancement. 

Hoe heb ik hier rekening mee gehouden?

**Progressive Enhancement:**

- Laag 1: Semantische HTML
- Laag 2: Basic CSS: visueel aantrekkelijk
- Laag 3: Usability: hover states, active states, etc
- Laag 4: Animations

**Toegankelijkheid**:

- Semantische HTML (goed voor bv screenreaders)
- Img's hebben een alt tekst
- Ik gebruik semantische HTML
- Er is hoge kleurcontrast

**Performance**:

- Zo compact mogelijke code
- Gebruik maken van componenten
- Ik gebruik lazy loading op img's
- Ik zet vaste `width` & `height` waardes op evt. afbeeldingen om CLS te voorkomen

## Responsive

De pagina is goed responsive voor telefoon (small screen), iPad (medium) en grotere schermen. Hierbij heb ik een mobile-first aanpak gehanteerd en dit genoteerd aan de hand van media-queries. Voor de kleine schermen zijn de cards en buttons onder elkaar in een `column`.  Dit is hoe het eruitziet op de verschillende schermen: 

## Mobile (small)

<img width="372" alt="Image" src="https://github.com/user-attachments/assets/1aaa20f2-b681-4170-9ade-344c65ca55e2" />
<img width="332" alt="Image" src="https://github.com/user-attachments/assets/f355059e-9427-4be8-a790-a567f73c7942" />


## Ipad (medium)

Hier is het verschil in de buttons in de card die naast elkaar komen te staan in plaats van onder elkaar, omdat er meer ruimte is. 

<img width="488" alt="Image" src="https://github.com/user-attachments/assets/ece6900f-a09e-4ee9-ad9b-92b6e6c00d74" />
<img width="222" alt="Image" src="https://github.com/user-attachments/assets/0a9a0abc-a4a9-45ad-817e-7bed641ce0b9" />

## Laptop (large)

Hier staan de cards naast elkaar en is de `width` ook wat breder. 

<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/db3eae9c-4d32-485a-9a5f-2ba9a147413f" />

## Accessibility

### Lighthouse

### Desktop 

Ik ben begonnen met het uitvoeren van de Lighthouse Test. De pagina heeft een score van 96 gekregen. 
<img width="550" alt="Image" src="https://github.com/user-attachments/assets/8baa5f36-9449-4aaf-bef9-013f8dbbe360" />

De verbeterpunten:

- Document doesn't have a `title` element

Wanneer ik dit kleine verbeterpuntje had aangepast is de accessibility nu 100! 🎉

### Mobile

Ik heb de Lighthouse Test ook uitgevoerd voor mobile, om te kijken of hier wat te verbeteren valt. Dat blijkt van niet, want de score hiervan is nu ook 100 :)

### Keyboard test

Vervolgens ben ik verder gegaan met de keyboard test. Hierbij heb ik gekeken of alle interactieve elementen te navigeren zijn met alleen een toetsenbord. Dit is tot nu toe wel het geval. De enige interactieve elementen op de pagina zijn de buttons. Deze hebben een `focus` state en krijgen een roze omlijning en worden iets groter zodat het duidelijk is dat de gebruiker daar nu is. 

<img width="146" alt="Image" src="https://github.com/user-attachments/assets/a9573535-2566-43ae-8b02-1d2167aec26e" />
<img width="173" alt="Image" src="https://github.com/user-attachments/assets/1278c275-e41c-46ce-a151-874c175160a8" />

### Screenreader
Ook heb ik met een screenreader geprobeerd om de pagina te navigeren. Ook dit is prima te volgen. Wel heb ik opgemerkt dat de 'tours' section geen heading heeft, waardoor er even wat verwarring ontstond. Dit heb ik opgelost door hier nog een `h1` aan toe te voegen en dit visueel te verbergen, maar niet voor de screenreader. 

``` HTML
<section class="tours">
  <h1>Amsterdam</h1>
  <TicketCard {itemCollection} />
</section>

```
``` CSS

.tours {
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}
```

### W3C HTML Validator

Er zijn een paar errors uitgekomen voor de pagina.

Twee van deze zijn op de `img` van de card component, hier zijn 2 attributen niet toegestaan om het zo te gebruiken:

- `Error: Attribute opacity not allowed on element`
- `Error: Attribute brdradius not allowed on element.`

Deze heb ik verwijderd (eigenlijk hadden ze geen toegevoegde waarde aangezien ze niks deden).

Hierna heb ik alleen nog een error/warning over de svg's, over een duplicate layer ID. Deze snap ik nog niet zo goed, dus hier moet ik nog even achteraan gaan. 


- `Warning: The first occurrence of ID Layer_1 was here.`
-` Error: Duplicate ID Layer_1.`

### Color Contrast Test

Voor de kleurcontrast test heb ik gekeken naar het contrast van de tekst button op de achtergrond hiervan. Ik kan concluderen dat de testresultaten best goed zijn, want ze hebben een goed contrast bij elke voorwaarden van de WCAG.

<img width="175" alt="Image" src="https://github.com/user-attachments/assets/cb2856ff-0d9a-4b20-be0d-106a8b11c9a8" />
<img width="175" alt="Image" src="https://github.com/user-attachments/assets/b711e99b-2397-4844-bee4-4fd4da5b4198" />


## Performance

De Performance Score is 78. Dit kan nog goed verbeterd worden. Het gaat voornamelijk om de Largest Contentful Paint (LCP) die best hoog is (4.4s). Het geldt voor de img in de cards. Die heeft geen `srcset` en het heeft wel `lazy loading`. 

De `srcset` zit in de `Image.svelte`

``` HTML

<picture>
  <source type="image/webp" media="(max-width: 375px)" srcset={smallWebp} />
  <source media="(max-width: 375px)" srcset={smallFallback} />
  <source type="image/webp" media="(max-width: 768px)" srcset={mediumWebp} />
  <source media="(max-width: 768px)" srcset={mediumFallback} />
  <source type="image/webp" media="(max-width: 1024px)" srcset={largeWebp} />
  <source media="(max-width: 1024px)" srcset={largeFallback} />
  <img
    src={largeFallback}
    {alt}
    {loading}
    style={`width: ${width}px; height: ${height}px; border-radius: ${brdRadius}; opacity: ${opacity}; view-transition-name: ${title};`}
  />
</picture>

```


## Progressive Enhancement 

Ook heb ik een PE test gedaan. Hierbij heb ik JavaScript uitgezet en gekeken of alles nog werkt. Dit is ook het geval, alles werkt zonder JavaScript. Ook in de verschillende browsers werkt alles naar behoren en ziet het er redelijk uit. Bijvoorbeeld op safari zijn de icons niet zichtbaar, maar dit is ook niet nodig aangezien er een titel in de button zit en het dus goed beschrijft wat de functie ervan is.
<img width="363" alt="Image" src="https://github.com/user-attachments/assets/70f1c0bb-3c5f-4f2d-a20f-1d0c8cbe3110" />


### Browser & device test

Hierboven heb ik mijn pagina getest op de browsers van mijn laptop. Echter was ik erg benieuwd hoe dit eruit zag op een oudere macbook Air (2015!). 

**Macbook Air 2015 - MacOS 10.13) - Chrome versie 116**

Je ziet het niet op de screenshot, alleen de roze kleur is wat feller dan wat er op de nieuwere schermen displayed wordt. Hiervoor is het kleurcontrast erg belangrijk, want anders kon je de tekst niet goed zien. Gelukkig is het kleurcontrast goed en is dit nog prima te lezen. 

<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/38383a07-31d1-4f52-9aa3-be102541e3c6" />

**Macbook Air 2015 - MacOS 10.13) - Safari versie 13.1.2**

In deze browser is er wel wat meer dingen niet goed, bijvoorbeeld dat alle images niet inladen, wel is de alt tekst zichtbaar. Maar ook dat er geen padding, margin, of gap tussen de cards en de buttons zitten. Ik heb alle properties gecheckt op CanIUse en alles wat ik gebruik ondersteunen deze browsers, dus ik moet hier nog even achteraan gaan. 

<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/8562e789-0621-4e64-9efe-a2ead45b5414" />

**Comment by zainablfz** [comment]
# User Test 07-02-2025

## Doelen van de test

Gebruikerservaring: Controleren of de card duidelijk, logisch en gebruiksvriendelijk is.

## Testscenario

Als gebruiker van WOGO wil ik bekijken welke tours WOGO aanbiedt en wil ik hier meer over lezen

## Taken

1. Naar de Tours pagina gaan
2. Bekijken welke tours er zijn 
3. Op de lees meer button kiezen om meer te weten 



# Testgebruiker 1 

De eerste testgebruiker is mijn vriendin, zij is het doelgroep van WOGO dus hiermee dacht ik dat dit een goede match is. Wanneer ik haar de website gaf, las zij eerst de hero content die met een animatie tevoorschijn kwam. Daarna is zij gelijk naar de navigatie bar gegaan om te kijken of er een pagina is die alle tours hebben. Die kon zij gemakkelijk vinden en daar klikte zij op. Hierna laadde de tours pagina in. Daarna vroeg zij zich gelijk af of zij nu de tours ziet van alle steden of alleen een bepaalde stad, aangezien de steden in de sectie daarboven zag. Na wat onduidelijkheid wilde zij gelijk de tours bekijken van Amsterdam dus zij klikte op deze stad. De tabfilter werkt niet, dus zij scrolde naar beneden om deze tours te bekijken. Hierna zag zij niet gelijk welke tour bij welke stad hoort. Ze zei wel dat ze het zou kunnen zien aan de locatie, bijvoorbeeld een tour stond er dat het in De Pijp - Jordaan was, dus zij nam aan dat dit in Amsterdam gevestigd is. Hierna is wilde zij meer weten over de tours, aangezien er niet veel informatie op de card staat. Hier zei ze dat alles duidelijk beschreven is over hoe het in zijn werking gaat. 

## Na de test

Wanneer de test klaar was heb ik haar nog een aantal vragen gesteld om de onduidelijkheden die naar voren kwam helder te maken voor mij om een oplossingsrichting uit te vinden. 

"**Waarom twijfelde je of de cards wel/niet gefilterd waren toen je op de pagina kwam?**" Waarop zij antwoordde: "Tja, ik kwam op de pagina en ik zag als eerst de steden staan, waarbij ik dacht dat er eerst de tours kwamen van Amsterdam, daarna Rotterdam, etc. 

Vervolgens vroeg ik door: "**Hoe denk jij dat dit duidelijker gedaan kan worden zodat jij/de gebruiker wel weet waar je je in bevindt in de tabfilter?**" Waarop zij antwoordde: "Ik denk dat het duidelijker zou zijn als er naast Amsterdam 'Alle' zou staan en het dan duidelijk is dat je je daarin bevindt en als je dan liever naar een specifieke stad wilt kijken dat je dan op een van die steden kan klikken."

Vervolgens vroeg ik door op haar antwoord wat voor haar dan duidelijk zou zijn: "**Hoe is het voor jou duidelijk waar je je in de filter bevindt?**" Hier antwoordde zij: "Uhm, misschien een vinkje? Of een andere kleur dan de rest zodat dit van hen onderscheidt"

Door deze vragen te stellen gaf mij dit een duidelijkere beeld hoe ik deze onduidelijkheid aan kan pakken en dit kan verbeteren 🍸

# Testgebruiker 2

De tweede testgebruiker is mijn vader. Hij is iets minder technisch, dus ik wilde hier kijken of alles duidelijk is. 
Ik liet hem beginnen op de homepagina, waar hij ook de hero content begon te lezen, hierna begon hij naar beneden te scrollen om te zoeken waar hij dit kan vinden, hij kwam eerst de reviews tegen, waarbij hij hardop zei "Ik wil eerst zien wat zij aanbieden, niet de reviews" Hierna is hij weer naar boven gescrold en is hij via de navigatie bar naar de tours pagina gegaan. Hierna begon hij hardop te lezen wat er op de tourcards stonden en heeft hij daarna op de 'Lees meer' button geklikt. Vervolgens heeft hij de content hierop gelezen, hij vond het erg fijn dat voordat de alinea's teksten begon er al een kleine samenvatting stond van de tour. 


## Na de test

Wanneer de test klaar was heb ik hem nog een aantal vragen gesteld om de onduidelijkheden die naar voren kwam helder te maken voor mij om een oplossingsrichting uit te vinden. 

"Waarom scrolde je eerst naar beneden ipv te kijken naar de navigatiebar wat je in tweede instantie deed?" Waarop hij antwoordde: "Ja dat doe ik eigenlijk altijd, maar ik wil niet al te veel naar beneden te scrollen om te vinden waar ik naar op zoek ben, daar heb ik geen geduld voor. Ik scroll ook niet zo snel, want ik doe het gewoon met het toetsenbord met dat pijltje naar beneden"

Waarop ik doorvroeg om dit punt te verbeteren: "Hoe zou deze onduidelijkheid verbeterd kunnen worden volgens jou?" Hier antwoordde hij: "Ik zou het fijner vinden om reviews en de cards sectie om te draaien, iig niet reviews als eerst. Naar mijn ervaring bekijk je die als laatst, of niet?
